src/Simulation/EntryPointDriver/Azure.cs

```diff
@@ -32,23 +32,45 @@ internal static class Azure
                 return 1;
             }
 
-            // TODO: Specify the number of shots. The IQuantumMachine interface should be updated.
-            var job = await machine.SubmitAsync(entryPoint.Info, entryPoint.CreateArgument(parseResult));
-            switch (settings.Output)
+            var input = entryPoint.CreateArgument(parseResult);
+            if (settings.DryRun)
             {
-                case OutputFormat.FriendlyUri:
-                    Console.WriteLine("Job submitted. To track your job status and see the results use:");
+                var (isValid, message) = machine.Validate(entryPoint.Info, input);
+                Console.WriteLine(isValid ? "✔️  The program is valid!" : "❌  The program is invalid.");
+                if (!string.IsNullOrWhiteSpace(message))
+                {
                     Console.WriteLine();
+                    Console.WriteLine(message);
+                }
+                return isValid ? 0 : 1;
+            }
+            else
+            {
+                var job = await machine.SubmitAsync(
+                    entryPoint.Info, input, new SubmissionContext { Shots = settings.Shots });
+                DisplayJob(job, settings.Output);
+                return 0;
+            }
+        }
+
+        /// <summary>
+        /// Displays the job using the output format.
+        /// </summary>
+        /// <param name="job">The job.</param>
+        /// <param name="format">The output format.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown if the output format is invalid.</exception>
+        private static void DisplayJob(IQuantumMachineJob job, OutputFormat format)
+        {
+            switch (format)
+            {
+                case OutputFormat.FriendlyUri:
                     // TODO: Show the friendly URI. The friendly URI is not yet available from the job.
-                    Console.WriteLine(job.Id);
```

Comment:
In this case maybe we could display something like "Not yet available" or "Available in a future release". This way we can let users know that this format option is not available in this release but it will be available in a future one. Otherwise, users that try this option early might not get why both the ID and the URI options display the same output.